keps/sig-node/4033-group-driver-detection-over-cri/README.md

```diff
@@ -0,0 +1,928 @@
+scenarios where the kubelet `--cgorup-driver` setting is incorrectly
+configured). With old versions of the container runtimes (that don't support
+the new field in the CRI API) the default behavior is not changed.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+<!--
+Describe the consequences on existing workloads (e.g., if this is a runtime
+feature, can it break the existing applications?).
+
+Feature gates are typically disabled by setting the flag to `false` and
+restarting the component. No other changes should be necessary to disable the
+feature.
+
+NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+-->
+
+Yes, through the feature gate.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Kubelet starts to use the cgroup driver instructed by the runtime. Potentially
+fixing a broken/misbehaving node if the kubelet cgroupDriver option (or
+`--cgroup-driver` flag) was incorrectly set.
+
+###### Are there any tests for feature enablement/disablement?
+
+<!--
+The e2e framework does not currently support enabling or disabling feature
+gates. However, unit tests in each component dealing with managing data, created
+with and without the feature, are necessary. At the very least, think about
+conversion tests if API types are being modified.
+
+Additionally, for features that are introducing a new API field, unit tests that
+are exercising the `switch` of feature gate itself (what happens if I disable a
+feature gate after having objects written with the new field) are also critical.
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+
+TBD.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+A rollout/rollback could fail only in the way it currently does: cgroup driver
+skew between CRI server (containe runtime) and Kubelet resulting in nodes going
+NotReady. This is only possible when the CRI server and Kubelet are not both
+upgraded to support the feature and are both not configured to agree on the
+CgroupDriver as they must be today.
+
+###### What specific metrics should inform a rollback?
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+N/A.
```

Comment:
so, if there were a bug in this code, then a bunch of nodes going NotReady would seem to be a reason to roll back

Reply:
Uh-oh, yes, this was an omission. Updated to:

```
Nodes being in NotReady state with kubelet logs indicating an error in the
RuntimeConfig CRI request, making kubelet fail to start.
```